### PR TITLE
Fix bgp_global errors.

### LINKED ISSUE #318: cisco.ios.ios_bgp_global error when using peer groups
##### SUMMARY
When gathering bgp from a device with peer groups configured, the task fails with a tracebook about address in the populate facts function.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
cisco.ios.bgp_global

##### ANSIBLE VERSION

```paste below
ansible 2.10.8
  config file = /home/<user>/configuration_management/ansible/ansible.cfg
  configured module search path = ['/home/<user>/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/<user>/.local/lib/python3.8/site-packages/ansible
  executable location = /home/<user>/.local/bin/ansible
  python version = 3.8.5 (default, Jan 27 2021, 15:41:15) [GCC 9.3.0]
```

##### CONFIGURATION

```paste below
DEFAULT_GATHERING = explicit
DEFAULT_HOST_LIST = ['/configuration_management/ansible/build']
DEFAULT_LOAD_CALLBACK_PLUGINS = True
DEFAULT_LOG_PATH = /var/log/ansible.log
DEFAULT_STDOUT_CALLBACK = yaml
HOST_KEY_CHECKING = False
```

##### OS / ENVIRONMENT

Cisco IOS Software [Gibraltar]
Version 16.12.05
ISR4331

##### STEPS TO REPRODUCE

Run task with cisco.ios.ios_bgp_global module and state = gathered

```yaml
    - name: Gather bgp config
      cisco.ios.ios_bgp_global:
        state: gathered
```

##### EXPECTED RESULTS

Return bgp configuration in yaml format

##### ACTUAL RESULTS

Task fails with traceback

```paste below
The full traceback is:
  File "/tmp/ansible_cisco.ios.ios_bgp_global_payload_xvpbu5qc/ansible_cisco.ios.ios_bgp_global_payload.zip/ansible_collections/ansible/netcommon/plugins/module_utils/network/common/facts/facts.py", line 133, in get_network_resources_facts
    inst.populate_facts(
  File "/tmp/ansible_cisco.ios.ios_bgp_global_payload_xvpbu5qc/ansible_cisco.ios.ios_bgp_global_payload.zip/ansible_collections/cisco/ios/plugins/module_utils/network/ios/facts/bgp_global/bgp_global.py", line 75, in populate_facts
    if temp["address"] != each["address"]:
```

### PR DESCRIPTION
**Pull request by dalrrard**
##### SUMMARY
Fix bgp_global error when peer-groups are defined.
The bgp_global module could only collect facts for ipv4 addresses.
Added the ability to work with ipv6 addresses and peer-groups.

Fix issue with peer-group neighbors not being found.

Fix #318 

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME
bgp_global

##### ADDITIONAL INFORMATION
The module would only accept IPv4 addresses as neighbors even though IPv6 addresses and peer-groups were defined in the argspec. Added path for all 3 to be created, and in the case of a peer-group, make sure that we define it in the config before using it in a neighbor.

### CODE REVIEW

File: plugins/module_utils/network/ios/argspec/bgp_global/bgp_global.py [hunk 1]
```diff
@@ -349,6 +349,9 @@ class Bgp_globalArgs(object):  # pylint: disable=R0903
                 "neighbor": {
                     "type": "list",
                     "elements": "dict",
+                    "mutually_exclusive": [
```
Comment (KB-perByte) [inline plugins/module_utils/network/ios/argspec/bgp_global/bgp_global.py:352]: Ansible won't recognize the `mutually_exclusive` key here as it is intended to work, rather it is recommended to use `choice` to handle the same. This is supposed to break parts of the execution.

**Review by KB-perByte — CHANGES_REQUESTED** [review]

**Comment by KB-perByte** [comment]
Hey @dalrrard, the bgp_global module is being considered for fresh code optimization. with some requested features and changes for the upcoming major release, We may consider the tagged issue with that. We might close this PR for the same.
Regards

**Comment by dalrrard** [comment]
No worries. I had a few more PRs coming in after this for that module for various things anyway. You're good to close this out if you want.

**Comment by KB-perByte** [comment]
Hey @dalrrard, closing this issue as most of the `bgp_global` related issues are collectively addressed in https://github.com/ansible-collections/cisco.ios/pull/513, you may check the PR and add any suggestion you have.
Regards

**Pull request closed without merging**